### ISSUE TITLE
the structure of note in vault dir are not provided for chat

### ISSUE THREAD

**Issue by r0lam** [reorproject/reor#100]
# issue:
  I tried to ask the chat for the structure of notes I writted(such as the title of markdown files , and the file tree of vault), but chat said these information are not provide.
  my prompt: please tell me what I have written in a tree structure.You can get references from the filename and dir-name in vault dir.
  The result is a list of the markdown head nodes, instead of the filename(titles of each note file) and  dir name.
  **It is nessary to provide the information mentioned above for the RAG system, I guess.**
# environment:
  ubuntu22-desktop
  openai-gpt4

**BTW, reor is an awsome project, respect for all contributors!**

**Comment by samlhuillier** [comment]
Yes unfortunately right now the chatbot only does a semantic search on your notes and can answer questions based on the contents of notes. It can't see anything beyond that like how the directory is structured.

We are working on moving towards much more agentic workflows like the one you describe where the model would make a function call to a filesystem API to display the contents of the directory. However, right now you can only ask questions based on what's in your notes. 

Can I ask what other things you'd want Reor to be able to answer/do for you? This will be very helpful for us as we build more agentic features.
